internal/ionoscloud/ionoscloud.go

```diff
@@ -0,0 +1,340 @@
+	logger := log.WithField(logFieldZoneID, zoneId).WithField(logFieldRecordName, name)
+	logger.Debug("get records from zone by name ...")
+	records, _, err := c.client.RecordsApi.RecordsGet(ctx).FilterZoneId(zoneId).FilterName(name).
+		FilterState(sdk.AVAILABLE).Execute()
+	if err != nil {
+		logger.Errorf("failed to get records from zone by name: %v", err)
+		return records, err
+	}
+	if records.HasItems() {
+		logger.Debugf("found %d records", len(*records.Items))
+	} else {
+		logger.Debug("no records found")
+	}
+	return records, nil
+}
+
+// GetZones client get zones method
+func (c *DNSClient) GetZones(ctx context.Context, offset int32) (sdk.ZoneReadList, error) {
+	log.Debug("get all zones ...")
+	zones, _, err := c.client.ZonesApi.ZonesGet(ctx).Offset(offset).Limit(zoneReadLimit).FilterState(sdk.AVAILABLE).Execute()
+	if err != nil {
+		log.Errorf("failed to get all zones: %v", err)
+		return zones, err
+	}
+	if zones.HasItems() {
+		log.Debugf("found %d zones", len(*zones.Items))
+	} else {
+		log.Debug("no zones found")
+	}
+	return zones, err
+}
+
+// CreateRecord client create record method
+func (c *DNSClient) CreateRecord(ctx context.Context, zoneId string, record sdk.RecordCreate) error {
+	recordProps := record.GetProperties()
+	logger := log.WithField(logFieldZoneID, zoneId).WithField(logFieldRecordName, *recordProps.GetName()).
+		WithField(logFieldRecordType, *recordProps.GetType()).WithField(logFieldRecordContent, *recordProps.GetContent()).
+		WithField(logFieldRecordTTL, *recordProps.GetTtl())
+	logger.Debugf("creating record ...")
+	if !c.dryRun {
+		recordRead, _, err := c.client.RecordsApi.ZonesRecordsPost(ctx, zoneId).RecordCreate(record).Execute()
+		if err != nil {
+			logger.Errorf("failed to create record: %v", err)
+			return err
+		}
+		logger.Debugf("created successfully record with id: '%s'", *recordRead.GetId())
+	} else {
+		logger.Info("** DRY RUN **, record not created")
+	}
+	return nil
+}
+
+// DeleteRecord client delete record method
+func (c *DNSClient) DeleteRecord(ctx context.Context, zoneId string, recordId string) error {
+	logger := log.WithField(logFieldZoneID, zoneId).WithField(logFieldRecordID, recordId)
+	logger.Debugf("deleting record: %v ...", recordId)
+	if !c.dryRun {
+		_, err := c.client.RecordsApi.ZonesRecordsDelete(ctx, zoneId, recordId).Execute()
+		if err != nil {
+			logger.Errorf("failed to delete record: %v", err)
+			return err
+		}
+		logger.Debug("record deleted successfully")
+	} else {
+		logger.Info("** DRY RUN **, record not deleted")
+	}
+	return nil
+}
+
+// Provider extends base provider to work with paas dns rest API
+type Provider struct {
+	provider.BaseProvider
+	client       DNSService
+	domainFilter endpoint.DomainFilter
+}
+
+// NewProvider returns an instance of new provider
+func NewProvider(domainFilter endpoint.DomainFilter, configuration *ionos.Configuration, dryRun bool) *Provider {
```

Comment:
I would expect to have more configurations in the future so I would already start introducing an vararg options pattern to keep the signature small and avoid breaking changes

Reply:
In this particular case every configuration is supposed to be in the `ionos.Configuration` struct. However, actually the flag `dryRun` is extra. But, - now when im looking on it -  I think this is wrong... will try to change this..